File: sstannouncer/app/src/main/java/sstinc/sstannouncer/Feed/Feed.java
package sstinc.sstannouncer.Feed;

import java.util.ArrayList;
import java.util.Date;

public class Feed {
    private Date lastChanged;
    private ArrayList<String> categories;
    private ArrayList<Entry> entries;

    public Feed(Date lastChanged, ArrayList<String> categories, ArrayList<Entry> entries) {
        this.lastChanged = lastChanged;
        this.categories = categories;
        this.entries = entries;
    }

    public Date getLastChanged() {
        return this.lastChanged;
    }

    public ArrayList<String> getCategories() {
        return this.categories;
    }

    public ArrayList<Entry> getEntries() {
        return this.entries;
    }
}

File: sstannouncer/app/src/main/java/sst/com/anouncements/feed/data/Feed.kt
package sst.com.anouncements.feed.data

import java.util.Date

data class Feed(
    val id: String,
    val entries: List<Entry>,
    val lastUpdated: Date,
    val categories: List<String>,
    val title: String,
    val subtitle: String
) {
    override fun equals(other: Any?): Boolean =
        other is Feed &&
        id == other.id &&
        lastUpdated.compareTo(other.lastUpdated) == 0

    override fun hashCode() = (id + lastUpdated).hashCode()
}